### PR TITLE
[PJRT C API] fix bug for CApi Bypass path

### PR DESCRIPTION
**Pull request by jzhoulon**
wrapped_ variable only works in c api bypass path(c wrapper path) since those struct(PJRT_Buffer, PJRT_Client, PJRT_Device) is only defined in the c api wrapper path but treated as opaque handle under real C API path.

### CODE REVIEW

**Comment by jzhoulon** [comment]
@jyingl3  can you help to take a look? thanks

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc [hunk 1]
```diff
@@ -406,14 +410,15 @@ StatusOr<std::unique_ptr<PjRtBuffer>> PjRtCApiClient::BufferFromHostBuffer(
     event_args.struct_size = PJRT_Event_OnReady_Args_STRUCT_SIZE;
     event_args.priv = nullptr;
     event_args.event = event.get();
-    event_args.user_arg = new std::function<void(PJRT_Error*)>(
```
Comment (jyingl3) [inline tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc:409]: Could you revert the format change?

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc [hunk 2]
```diff
@@ -1083,7 +1093,7 @@ StatusOr<std::unique_ptr<PjRtBuffer>> PjRtCApiBuffer::CopyToDevice(
         literal_pointer->shape().element_type(),
         literal_pointer->shape().dimensions(), byte_strides,
         PjRtClient::HostBufferSemantics::kZeroCopy,
-        [literal{std::move(literal)}]() { /* frees literal */ }, dst_device);
```
Comment (jyingl3) [inline tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc:1086]: Could you revert the format change?

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc [hunk 3]
```diff
@@ -1118,7 +1128,7 @@ void PjRtCApiBuffer::MakePromiseTrackEvent() {
   args.priv = nullptr;
   args.event = GetReadyEvent();
   args.user_arg = new std::function<void(PJRT_Error*)>(
-      [promise = readiness_promise_, api](PJRT_Error* error) -> void {
```
Comment (jyingl3) [inline tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc:1121]: Could you revert the format change?

**Review by jyingl3 — COMMENTED** [review]
Thanks for fixing the issue! The change looks good to me. Just minor comments about reverting the format change.

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc (hunk 3 above)
Comment (jyingl3) [reply tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc:1121]: @skye FYI

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc (hunk 1 above)
Comment (jzhoulon) [reply tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc]: done

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc (hunk 2 above)
Comment (jzhoulon) [reply tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc]: done

File: tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc (hunk 3 above)
Comment (jzhoulon) [reply tensorflow/compiler/xla/pjrt/pjrt_c_api_client.cc]: done.

**Comment by jzhoulon** [comment]
@jyingl3 I have revert the clang format , can you help to have a look again? thanks

**Review by jyingl3 — APPROVED** [review]
Thanks for fixing it!

**Comment by jzhoulon** [comment]
@jyingl3 I saw the PR is stuck with some internal checks Failed, if there is something need to be fixed in my PR, please let me know:)

**Comment by jyingl3** [comment]
Sorry it took a while. But it is in now!

**Pull request merged**